src/lib/components/Sidebar.svelte

```diff
@@ -37,8 +111,15 @@
       </a>
       {#if chats.length}
         <ul class="overflow-auto grow flex flex-col gap-2 pt-2">
-          {#each chats as chat (chat.id)}
-            <ChatLink {chat} {user} />
+          {#each Object.entries(chatsGroupedByTime) as [interval, chats]}
+            {#if chats.length > 0}
+              {#each chats as chat}
+                {#if renderTitle(interval)}
+                  <p class="flex-none text-xs text-gray-600">{getTitle(interval)}</p>
+                {/if}
+                <ChatLink {chat} {user} />
+              {/each}
+            {/if}
```

Comment:
I don't think you need the `renderTitle` function or the `if`-statement using it. If you move the title outside of the inner `#each`, the title should end up where it belongs, right?
```svelte
{#each Object.entries(chatsGroupedByTime) as [interval, chats]}
  {#if chats.length > 0}
    <p class="flex-none text-xs text-gray-600">{getTitle(interval)}</p>
    {#each chats as chat}
      <ChatLink {chat} {user} />
    {/each}
  {/if}
{/each}
```

Reply:
Yeees you're right!!